src/OpenTelemetry.Contrib.Instrumentation.Remoting/RemotingInstrumentation.cs

```diff
@@ -0,0 +1,45 @@
+﻿// <copyright file="RemotingInstrumentation.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using OpenTelemetry.Contrib.Instrumentation.Remoting.Implementation;
+using RemotingContext = System.Runtime.Remoting.Contexts.Context;
+
+namespace OpenTelemetry.Trace
+{
+    internal class RemotingInstrumentation : IDisposable
+    {
+        public RemotingInstrumentation(RemotingInstrumentationOptions options)
+        {
+            // See https://docs.microsoft.com/en-us/dotnet/api/system.runtime.remoting.contexts.context.registerdynamicproperty?view=netframework-4.8
+            // This will register our dynamic sink to listen to all calls leaving or entering
+            // current AppDomain.
+            RemotingContext.RegisterDynamicProperty(
```

Comment:
Safe to be called multiple times? Users can register multiple TracerProviders so might need to check static variable here or something to make it safe?

Reply:
Nope, that would have been a bug - thanks!

I've added checks on both Register and Unregister. I am not sure if anyone will ever call that from different thread, but added a `lock` just in case. I imagine it will be called so rarely that a big lock won't be an issue?